server/user.go

```diff
@@ -44,6 +44,7 @@ func (c *Connection) JiraAccountID() types.ID {
 
 type ConnectionSettings struct {
 	Notifications bool `json:"notifications"`
+	Watching      bool `json:"watching"`
```

Comment:
@mickmister I've pushed a new commit for this change, could you please check it?

Reply:
@DHaussermann @aaronrothschild Please see my comment above

- If the current settings for a user has notifications "on" before this PR, should we assume the user also wants reporter and watcher notifications when we introduce the changes from this PR? Should we tell the user how to change their settings? Maybe we only tell them how to change their settings the next time they get a notification. We tell them how to change their settings, and don't remind them again.

- Should we tell the user the _reason_ they are getting a notification? It would be one of the following reasons:
    - Mentioned
    - Assignee
    - Reporter
    - Watching